lambdas/process-mobile-app-callback/src/main/java/uk/gov/di/ipv/core/processmobileappcallback/ProcessMobileAppCallbackHandler.java

```diff
@@ -144,64 +120,57 @@ private MobileAppCallbackRequest parseCallbackRequest(APIGatewayProxyRequestEven
         }
     }
 
-    private void validateCallback(MobileAppCallbackRequest callbackRequest)
+    private JourneyResponse validateCallback(MobileAppCallbackRequest callbackRequest)
             throws InvalidMobileAppCallbackRequestException, IpvSessionNotFoundException,
                     ClientOauthSessionNotFoundException, InvalidCriResponseException {
-        // Validate callback sessions
-        validateSessionId(callbackRequest);
+        // Validate CRI state
+        var criState = callbackRequest.getState();
+        if (StringUtils.isBlank(criState)) {
+            throw new InvalidMobileAppCallbackRequestException(ErrorResponse.MISSING_OAUTH_STATE);
+        }
+
+        var criOAuthSessionItem = criOAuthSessionService.getCriOauthSessionItem(criState);
+        if (criOAuthSessionItem == null) {
+            throw new InvalidMobileAppCallbackRequestException(ErrorResponse.INVALID_OAUTH_STATE);
+        }
+
+        // Check IPV session
+        var ipvSessionId = callbackRequest.getIpvSessionId();
+        if (StringUtils.isBlank(ipvSessionId)) {
+            auditService.sendAuditEvent(
+                    AuditEvent.createWithDeviceInformation(
```

Comment:
with/without device information should probably be consistent between the two events.

Reply:
This is copying what already existed for the existing DCMAW audit events